src/lib389/lib389/replica.py

```diff
@@ -822,6 +822,25 @@ def demote(self, suffix, newrole):
             raise ValueError('Failed to update replica: ' + str(e))
 
 
+class _rdict(dict):
+    """A dict whose key is a Normalized Replica ID
+    """
+
+    def __init__(self, log):
+        self.d = {}
+        self._log = log
+
+    def __getitem__(self, key):
+        nkey = int(key)
+        self._log.debug(f'MYDBG: __getitem__ {key} {nkey} {self.__getitem__(nkey)}')
+        return self.__getitem__(nkey)
+
+    def __setitem__(self, key, value):
+        nkey = int(key)
+        self._log.debug(f'MYDBG: __setitem__ {key} {nkey} {value}')
+        self.d.__setitem__(nkey, value)
```

Comment:
I am a bit concerned with the dict implementation.

At least, I think, it needs to use `super().` in the returns and in the init - `super().__init__()`

Also, the debug messages with `MYDBG` look strange. If we want to keep the messages, I'd at least change the `MYDBG` part:)

Reply:
Also, maybe rename it to something like `NormalizedRidDict`?

Reply:
Ok with all your remarks. 
In fact the way I wrote _rdict, it is not really a dict but an object containing a dict. It works because we only get/set elements in this dict but it is clearly inconsistent ...

I will change that and also add a testcase to check the  NormalizedRidDict behavior